bentoml/onnx.py

```diff
@@ -274,28 +277,34 @@ def _setup(self) -> None:  # type: ignore[override]
             session_options=self._session_options,
             model_store=self._model_store,
         )
-        self._infer_func = getattr(self._model, "run")
+        raw_infer_func = getattr(self._model, "run")
+        self._infer_func = functools.partial(raw_infer_func, **self._partial_kwargs)
 
     def _run_batch(
         self,
-        *args: t.Union["np.ndarray", "pd.DataFrame"],
-        **kwargs: t.Any,
+        *args: t.Union[np.ndarray, "pd.DataFrame"],
     ) -> t.Any:
-        params = Params[t.Union["np.ndarray", "pd.DataFrame"]](*args, **kwargs)
-        if isinstance(params.sample, np.ndarray):
-            params = params.map(lambda i: i.astype(np.float32))
-        elif isinstance(params.sample, pd.DataFrame):
-            params = params.map(lambda i: i.to_numpy())
-        else:
-            raise TypeError(
-                f"`_run_batch` of {self.__class__.__name__} only takes "
-                "`numpy.ndarray` or `pd.DataFrame` as input parameters"
-            )
+        params = Params[t.Union[np.ndarray, "pd.DataFrame"]](*args)
+
+        def _mapping(item) -> t.Any:
+            if isinstance(item, np.ndarray):
+                item = item.astype(np.float32)
```

Comment:
I was thinking when writing this line is to allow users to define given type for numpy here. But I'm not entirely sure if that is necessary since I'm unsure about some usecases.

Reply:
If we add Tensors, do we still want to do conversion here? Or should we just check if the input data is types we support?

Reply:
would onnx accept tensor directly?

Reply:
no tensor so I think current codes is ok